cpp/src/ray/config_internal.cc

```diff
@@ -145,11 +147,16 @@ void ConfigInternal::Init(RayConfig &config, int argc, char **argv) {
   }
 };
 
-void ConfigInternal::SetRedisAddress(const std::string address) {
+void ConfigInternal::SetBootstrapAddress(const std::string &address) {
   auto pos = address.find(':');
   RAY_CHECK(pos != std::string::npos);
-  redis_ip = address.substr(0, pos);
-  redis_port = std::stoi(address.substr(pos + 1, address.length()));
+  if (::RayConfig::instance().bootstrap_with_gcs()) {
+    gcs_ip = address.substr(0, pos);
+    redis_port = std::stoi(address.substr(pos + 1, address.length()));
```

Comment:
This should be `gcs_port`? Or maybe we can use `port` or `bootstrap_port` for both `redis_port` and `gcs_port`?